### DISCUSSION TITLE
Using plotstyles with drawing addon

### DISCUSSION THREAD

**Discussion by akantifirst** [mozman/ezdxf#607] [Q&A]
I was able to find in the docs the corresponding section, but not sure how to implement it. When exporting .dxf file to .pdf with matplotlib backend, I get square line endings. It is not much of an issue, but may be somebody could give me an idea on how to implement CTB plotstyles to make line endings round. The picture illustrates what I mean:
![Screenshot_20220108-211134_Xodo Docs.jpg](https://user-images.githubusercontent.com/82590770/148658696-e9d0129c-bca2-42e5-8809-53e08fa555fa.jpg)

**Comment by mbway** [comment]
if the lines are joined in the dxf file then they should be drawn as a single path
https://github.com/mozman/ezdxf/blob/3fdba53e9ea058bbdf0a8a9228fbc662e44bfe0b/src/ezdxf/addons/drawing/matplotlib.py#L128
but from the screenshot it looks like they are rendered as separate line segments.

If you want to give all line segments rounded caps you could try passing `solid_capstyle='round', dash_capstyle='round'` here
https://github.com/mozman/ezdxf/blob/3fdba53e9ea058bbdf0a8a9228fbc662e44bfe0b/src/ezdxf/addons/drawing/matplotlib.py#L492-L507

https://matplotlib.org/stable/api/_as_gen/matplotlib.lines.Line2D.html

**Reply by akantifirst** [reply]
Hello Matt, thank you for the advice. I'm struggling to implement it like this, but the generated pdf still looks same as with default MatplotlibLineRenderer.
```Python
class CustomBackend(MatplotlibBackend):
        ...
        self._line_renderer: CustomLineRenderer


class CustomLineRenderer(MatplotlibLineRenderer):

    def __init__(self, config: Configuration, ax: plt.Axes, solid_only: bool):
        super().__init__(config, ax)
        self._solid_only = solid_only

    def draw_line(
            self,
            start: Vec3,
            end: Vec3,
            properties: Properties,
            z: float
    ):
        self.ax.add_line(
            Line2D(
                (start.x, end.x),
                (start.y, end.y),
                linewidth=self.lineweight(properties),
                linestyle="solid"
                if self._solid_only
                else self.linetype(properties),
                color=properties.color,
                zorder=z,
                solid_capstyle='round',
                dash_capstyle='round'
            )
        )
```
Can you please show me how to implement it correctly?

**Reply by mbway** [reply]
I double checked and setting `solid_capstyle` should definitely create rounded edges. For example:
```python
from matplotlib import pyplot as plt
from matplotlib.lines import Line2D


fig, ax = plt.subplots()
ax.add_line(Line2D((0, 1), (0, 1), linewidth=5, solid_capstyle='round'))
ax.autoscale(True)
plt.show()
```
perhaps your lines are actually paths which would have a different code path and not go through `draw_line()`?

you could try creating a custom backend which uses the default `draw_path` implementation which breaks them up into line segments then calls `draw_line()`
https://github.com/mozman/ezdxf/blob/3fdba53e9ea058bbdf0a8a9228fbc662e44bfe0b/src/ezdxf/addons/drawing/backend.py#L139-L156
